# CAT15 config

topics:
  input:
    lidar: /velodyne_points
    imu: /EL/Sensors/vectornav/IMU
  # output:

num_threads: 20 # OpenMP threads (will be equal to $(nproc) if it is set higher)

sensor_type: 1  # LiDAR type (0: OUSTER \ 1: VELODYNE \ 2: HESAI \ 3: LIVOX)

debug: true     # save useful intermediate pcl (pc2match, deskewed...)
verbose: true   # print debug/performance board

estimate_extrinsics: true   # continuous estimation of LiDAR-IMU extrinsics 

calibration:    # automatic IMU calibration
  gravity_align: false
  accel: true
  gyro: true
  time: 3.0

extrinsics:
  baselink2imu:
    t: [ 0.0, 0.0, 0.0 ]
    R: [ 1.,  0.,  0.,
          0.,  1.,  0.,
          0.,  0.,  1. ]
  baselink2lidar:
    t: [ 1.25,  0.,  0. ]
    R: [ 0.99,  0.,  0.09,
          0.,  1.,  0.,
        -0.09,  0.,  0.99 ]

intrinsics:
  accel:
    bias: [ 0.0, 0.0, 0.0 ]
    sm:   [ 1.,  0.,  0.,
            0.,  1.,  0.,
            0.,  0.,  1. ]
  gyro:
    bias: [ 0.0, 0.0, 0.0 ]

filters:
  cropBox:
    active: true
    box:
      min: [ -2.0, -2.0, -1.0 ]
      max: [ 2.0, 2.0, 1.0 ]

  voxelGrid:
    active: true
    leafSize: [ 0.25, 0.25, 0.25 ]

iKFoM:
  MAX_NUM_ITERS: 3
  NUM_MATCH_POINTS: 5
  MAX_DIST_PLANE: 2.0
  PLANES_THRESHOLD: 5.0e-2
  LIMITS: 0.001
  covariance:
    gyro: 6.01e-4 
    accel: 1.53e-2
    bias_gyro: 1.54e-5
    bias_accel: 3.38e-4
